tools/swift-inspect/Sources/swift-inspect/Process.swift

```diff
@@ -43,6 +43,59 @@ private func refersToSelf(_ str: String) -> Bool {
   // No match.
   return false
 }
+
+internal func getRemoteProcess(processId: ProcessIdentifier,
+                               options: UniversalOptions) -> (any RemoteProcess)? {
+  return DarwinRemoteProcess(processId: processId,
+                             forkCorpse: options.forkCorpse)
+}
+
+internal func getProcessName(processId: ProcessIdentifier) -> String? {
+  var info = proc_bsdinfo()
+  let bsdinfoSize = Int32(MemoryLayout<proc_bsdinfo>.stride)
+  let size = proc_pidinfo(processId, PROC_PIDTBSDINFO, 0, &info, bsdinfoSize)
+  if (size != bsdinfoSize) {
+    return nil
+  }
+  let processName = withUnsafePointer(to: info.pbi_name) {
+    $0.withMemoryRebound(to: CChar.self, capacity: MemoryLayout<CChar>.stride) {
+      String(cString: $0)
+    }
+  }
+  return processName
+}
+
+internal func getAllProcesses(options: UniversalOptions) -> [ProcessIdentifier]? {
+  var ProcessIdentifiers = [ProcessIdentifier]()
+  let kinfo_stride = MemoryLayout<kinfo_proc>.stride
+  var bufferSize: Int = 0
+  var name: [Int32] = [CTL_KERN, KERN_PROC, KERN_PROC_ALL]
+
+  guard sysctl(&name, u_int(name.count), nil, &bufferSize, nil, 0) == 0 else {
+    return nil
+  }
+  var count = bufferSize / kinfo_stride
+  var buffer = Array(repeating: kinfo_proc(), count: count)
+  guard sysctl(&name, u_int(name.count), &buffer, &bufferSize, nil, 0) == 0 else {
+    return nil
+  }
+
+  let myPid = getpid()
+  for kinfo in buffer.sorted { first, second in
+      first.kp_proc.p_pid > second.kp_proc.p_pid
+    } {
```

Comment:
This construct with a trailing closure followed by a for/in closure is a little strange to read. I suggest making a temporary variable for the sorted array instead, like:

```
let sorted = buffer.sorted { $0.kp.proc.p_pid > second.kp_proc.p_pid }
for kinfo in sorted { ...
```